modules/mcuboot/CMakeLists.txt

```diff
@@ -2,6 +2,12 @@ add_subdirectory(${ZEPHYR_MCUBOOT_MODULE_DIR}/boot/bootutil/zephyr
                  ${CMAKE_CURRENT_BINARY_DIR}/boot/bootutil/zephyr
 )
 
+if (CONFIG_MCUBOOT)
+  if (CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY)
```

Comment:
The reason this is not in a single if `CONFIG_MCUBOOT AND CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY` is that this is updated in the next commit.